httpserver/request_handler.cpp

```diff
@@ -76,9 +77,34 @@ Response RequestHandler::handle_wrong_swarm(const user_pubkey_t& pubKey) {
                     ContentType::json};
 }
 
+std::string computeMessageHash(const std::string& timestamp,
+                               const std::string& ttl,
+                               const std::string& recipient,
+                               const std::string& data) {
+
+    std::string payload;
+    payload.reserve(timestamp.size() + ttl.size() + recipient.size() +
+                    data.size());
+    payload += timestamp;
+    payload += ttl;
+    payload += recipient;
+    payload += data;

```

Comment:
`openssl` already provides an interface for piecewise SHA512 that avoids the need to copy the strings around into a single buffer:
```C++
SHA512_CTX ctx;
SHA512_Init(&ctx);
for (const auto* s : {timestamp, ttl, recipient, data})
    SHA512_Update(&ctx, s->data(), s->size());
unsigned char hash[SHA512_DIGEST_LENGTH];
SHA512_Final(hash, &ctx);
return oxenmq::to_hex(std::begin(hash), std::end(hash));
```

Reply:
Updated